Document the format of the policies field in flow logs. (#618)
tigera/docs: calico-cloud/visibility/elastic/flow/datatypes.mdx

| `packets_out`                     | long              | Number of outgoing packets since the last export.                                                                                                                                                                                                                                                                                                    |
| `proto`                           | keyword           | Protocol.                                                                                                                                                                                                                                                                                                                                            |

> | `policies`                        | array of keywords | List of policies that interacted with this flow; the format is described [below](#format-of-the-policies-field).                                                                                                                                                                                                                                                |

Reviewer: ```suggestion
| `policies`                        | array of keywords | List of policies that interacted with this flow. See [Format of the policies field](#format-of-the-policies-field).                                                                                                                                                                                                                                                |
```

| `tcp_unrecovered_to`              | long              | Total unrecovered timeouts. This field is only available if flowLogsEnableTcpStats is enabled                                                                                                                                                                                                                                                        |

### Format of the policies field

The `policies` field contains a comma-delimited list of policy rules that matched the flow.  Each entry in the
list has the following format:

```
<index>|<tier name>|<policy name>|<action>|<rule index>
```

Where,

* `<index>` numbers the order in which the rules were hit, starting with `0`.
  :::tip
  Sort the entries of the list by the `<index>` to see the order that rules were hit.
  :::

* `<tier name>` is the name of the policy tier containing the policy, or `__PROFILE__` for a rule derived from a

>   `Profile` resource (this is the internal datatype used to represent a Kubernetes Namespace and its associated

Reviewer: ```suggestion
  `Profile` resource (this is the internal datatype used to represent a Kubernetes namespace and its associated
```

  "default allow" rule).
* `<policy name>` is the name of the policy/profile; its format depends on the type of policy:

>   * `<tier>.<name>` for {{prodname}} `GlobalNetowrkPolicy`.

Reviewer: ```suggestion
  * `<tier>.<name>` for {{prodname}} `GlobalNetworkPolicy`.
```

  * `<namespace>/<name>` for Kubernetes `NetworkPolicy`.
  * `<namespace>/<tier>.<name>` for {{prodname}} `NetworkPolicy`.

* `<action>` is the action performed by the rule; one of `allow`, `deny`, `pass`.
* `<rule index>` if non-negative, is the index of the rule that was matched within the policy, starting with 0.
  Otherwise, a special value:

>   * `-1` means that the flow didn't match any policy; it hit the default action for the tier.  In this case, the

Reviewer: ```suggestion
  * `-1` means that the flow didn't match any policy. It hit the default action for the tier.  In this case, the
```
We're not fancy enough for semicolons! ;)

    policy name is an arbitrary policy that was active for the endpoint.

>   * `-2` means "unknown"; the rule index was not recorded.

Reviewer: ```suggestion
  * `-2` means "unknown". The rule index was not recorded.
```

### Format of the policies field

The `policies` field contains a comma-delimited list of policy rules that matched the flow.  Each entry in the
list has the following format:

```
<index>|<tier name>|<policy name>|<action>|<rule index>
```

Where,

* `<index>` numbers the order in which the rules were hit, starting with `0`.
  :::tip
  Sort the entries of the list by the `<index>` to see the order that rules were hit.
  :::

* `<tier name>` is the name of the policy tier containing the policy, or `__PROFILE__` for a rule derived from a
  `Profile` resource (this is the internal datatype used to represent a Kubernetes namespace and its associated
  "default allow" rule).
* `<policy name>` is the name of the policy/profile; its format depends on the type of policy:

  * `<tier>.<name>` for {{prodname}} `GlobalNetworkPolicy`.

>   * `<namespace>/<name>` for Kubernetes `NetworkPolicy`.

Reviewer 2: I haven't tested and I've forgotten where the code lives, but I thought these were actually <namespace>/knp.default.<name>?  Could be wrong though - just trying to remember.

Author: Yes, looks like it.

  * `<namespace>/<tier>.<name>` for {{prodname}} `NetworkPolicy`.

  Staged policy names are prefixed with "staged:".

* `<action>` is the action performed by the rule; one of `allow`, `deny`, `pass`.
* `<rule index>` if non-negative, is the index of the rule that was matched within the policy, starting with 0.
  Otherwise, a special value:

>   * `-1` means that the flow didn't match any policy. It hit the default action for the tier.  In this case, the

Reviewer 2: I wouldn't word this as didn't match any policy.  I try and use "match" to refer to rules only, and "apply" or "select" for policies.

Something like "means the reporting endpoint was selected by the policy but no rule matched and it hit the default action for the tier. ..." maybe?  Or maybe even replace "hit the default action for the tier" with "was denied at the end of the tier", since the default action isn't configurable.

Author: I could be wrong, but I think the default action can be "pass" if you have a staged tier?
